awx/main/migrations/0164_remove_inventorysource_update_on_project_update.py

```diff
@@ -1,24 +1,13 @@
 # Generated by Django 3.2.13 on 2022-06-21 21:29
 
 from django.db import migrations
-import logging
-
-logger = logging.getLogger("awx")
 
 
 def forwards(apps, schema_editor):
     InventorySource = apps.get_model('main', 'InventorySource')
     sources = InventorySource.objects.filter(update_on_project_update=True)
-    for src in sources:
-        if src.update_on_launch == False:
-            src.update_on_launch = True
-            src.save(update_fields=['update_on_launch'])
-            logger.info(f"Setting update_on_launch to True for {src}")
-        proj = src.source_project
-        if proj and proj.scm_update_on_launch is False:
-            proj.scm_update_on_launch = True
-            proj.save(update_fields=['scm_update_on_launch'])
-            logger.warning(f"Setting scm_update_on_launch to True for {proj}")
+    sources.update(update_on_launch=True)
+    sources.filter(source_project__isnull=False).update(scm_update_on_launch=True)
```

Comment:
For my purposes of reading this, let me sub in the prior definition of the `sources` QuerySet

```python
InventorySource.objects.filter(update_on_project_update=True).update(scm_update_on_launch=True)
```

This is attempting to replace `proj.save(update_fields=['scm_update_on_launch'])`... and put in those terms the problem becomes kind of obvious. This was making a change to _projects_, but you're using an InventorySource QuerySet. I think you need to start the QuerySet over, starting from `Project.objects`

Reply:
Oops.  Good catch.

Reply:
@AlanCoding ok, I think I have it correct now, let me know.